site/src/app/v2/test-eligibilite/components/step-two-forms/YoungMsaForm.tsx

```diff
@@ -0,0 +1,198 @@
+import Input from '@codegouvfr/react-dsfr/Input';
+import { ChangeEvent, FormEvent, useRef, useState } from 'react';
+import {
+  ConfirmResponseBody,
+  ConfirmResponseError,
+  SearchResponseBodyItem,
+  YoungMsaInputsState,
+} from 'types/EligibilityTest';
+import { convertDate, mapper } from '../../helpers/helper';
+import FormButton from './FormButton';
+import CommonMsaInputs from './common-msa-inputs/CommonMsaInputs';
+import ErrorAlert from '../error-alert/ErrorAlert';
+
+const initialInputsState: YoungMsaInputsState = {
+  recipientLastname: { state: 'default' },
+  recipientFirstname: { state: 'default' },
+  recipientBirthDate: { state: 'default' },
+  recipientBirthCountry: { state: 'default' },
+};
+
+interface Props {
+  eligibilityDataItem: SearchResponseBodyItem;
+  onDataRecieved: (data: ConfirmResponseBody) => void;
+}
+
+const YoungMsaForm = ({ eligibilityDataItem, onDataRecieved }: Props) => {
+  const formRef = useRef<HTMLFormElement>(null);
+  const [inputStates, setInputStates] = useState<YoungMsaInputsState>(initialInputsState);
+  const [isFormDisabled, setIsFormDisabled] = useState<boolean>(false);
+
+  const [error, setError] = useState<string | null>();
+
+  const isFormValid = (formData: FormData): { isValid: boolean; states: YoungMsaInputsState } => {
+    let isValid = true;
+
+    const fieldNames = Object.keys(inputStates) as (keyof YoungMsaInputsState)[];
+    const states = structuredClone(initialInputsState);
+
+    fieldNames.forEach((fieldname) => {
+      states[fieldname] = { state: 'default' };
+    });
+
+    fieldNames.forEach((fieldName) => {
+      const value = formData.get(fieldName);
+
+      if (!value) {
+        states[fieldName] = { state: 'error', errorMsg: mapper[fieldName] };
+        isValid = false;
+      }
+    });
+
+    return { isValid, states };
+  };
+
+  const requestPassSportCode = (): Promise<{ status: number; body: unknown }> => {
+    const domain = process.env.NEXT_PUBLIC_LCA_API_URL;
+
+    if (!domain) {
+      throw new Error('Error: NEXT_PUBLIC_LCA_API_URL is not set');
```

Comment:
On s'était pas dit que ce call de récupération de code devait être coté BE ? (J'ai peut-être raté un épisode)

Reply:
Ca sera fait dans une autre PR

Reply:
yes, une autre pr